packages/plugin-hardhat/src/utils/deploy.ts

```diff
@@ -9,7 +11,24 @@ export async function deploy(
   factory: ContractFactory,
   ...args: unknown[]
 ): Promise<Required<Deployment & DeployTransaction>> {
-  const { address, deployTransaction } = await factory.deploy(...args);
+  const contractInstance = await factory.deploy(...args);
+  const deployTransaction = contractInstance.deployTransaction;
```

Comment:
Shorter syntax that I prefer but let me know if you disagree:

```suggestion
  const { deployTransaction } = contractInstance;
```